matlab/src/matlab/+arrow/array.m

```diff
@@ -0,0 +1,67 @@
+% Licensed to the Apache Software Foundation (ASF) under one or more
+% contributor license agreements.  See the NOTICE file distributed with
+% this work for additional information regarding copyright ownership.
+% The ASF licenses this file to you under the Apache License, Version
+% 2.0 (the "License"); you may not use this file except in compliance
+% with the License.  You may obtain a copy of the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+% implied.  See the License for the specific language governing
+% permissions and limitations under the License.
+
+function arrowArray = array(data, opts)
+    arguments
+        data
+        opts.InferNulls(1, 1) logical = true
+        opts.Valid
+    end
+
+    data = convertCellstrToStrings(data);
+    classname = string(class(data));
+    args = namedargs2cell(opts);
+
+    switch (classname)
+        case "logical"
+            arrowArray = arrow.array.BooleanArray.fromMATLAB(data, args{:});
+        case "int8"
```

Comment:
This is a minor comment, but in a lot of other places we have been ordering this list of types as `uint8`, `uint16`, ... rather than "interleaving" the signed and unsigned integer types.

Reply:
That's a good point. I'll reorder the cases.